### PR TITLE
[FIX] website: prevent upsert crash if GEOIP country not found in odoo

### PR DESCRIPTION
**Pull request by rdeodoo**
It was reported the following error:
```
psycopg2.errors.DatatypeMismatch: column "country_id" is of type integer
                                  but expression is of type boolean
LINE 9:                 false, 1, NULL, 4, 4,
```

The root cause is unknown, it's not yet sure how this is happening exactly.
This commit is then fixing naively the consequence.

Somehow, GEOIP is returning a country code which does not exists in Odoo for this error to appear. It's normally not possible.

There is multiple possible case that would lead to this error:
- The user modified the `code` field of a country in his DB (unlikely)
- The user deleted a country in his DB (unlikely)
- The GEOIP database is returning a country which does not exists in Odoo. This is the most probably case, even if the latest up to date `GeoLite 2 City` seems to have the exact same country code list as in Odoo 16.0 (250 exact same ones).

### CODE REVIEW

File: addons/website/models/website_visitor.py [hunk 1]
```diff
@@ -209,6 +209,11 @@ def _upsert_visitor(self, access_token, force_track_values=None):
         country_id = request.env['res.country'].sudo().search([
             ('code', '=', country_code)
         ], limit=1).id if country_code else None
```
Comment (mart-e) [inline addons/website/models/website_visitor.py:211]:
```suggestion
        ], limit=1).id or None if country_code else None
```

Comment (rdeodoo) [reply addons/website/models/website_visitor.py:211]: That would be the obvious fix (or below: `'country_id': country_id,` -> `'country_id': country_id or None,`
But I'd like to keep as is, encapsulated in its own condition with an easy to understand comment, given the weirdness of the fix. It's yet unsure if someone messed with this GEOIP db or his Odoo countries.

**Comment by qsm-odoo** [comment]
Moved it back to draft as in the end we'll move the country search inside the SQL query (although I don't know if you planned to do it for master only @rdeodoo you'll tell me).

**Comment by rdeodoo** [comment]
I moved it in the upsert, and yes for 16.0

Perf wise, it doesn't seems to alter at all the request speed, plus it removes a query when geoip is enabled + it indirectly fixes the reported issue.

(Other solution was to go with the `False -> None` check + 
```py
@tools.ormcache('code')
 def _country_get_id(self, code):
    return self.with_context(active_test=True).search([('code', '=', code)]).id
 ```
 as we do have in `res.lang` but the upsert fix should be good enough)

**Review by qsm-odoo — APPROVED** [review]
Tested, seems nice :+1: Maybe we could improve someday to log / gather somehow what country code are failing (just a country_code field on the visitor model maybe although I guess we don't want an extra field just for that strange mystery).

@robodoo r+

**Pull request closed without merging**